File: 125-valid-palindrome/valid-palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function(s) {
    let filtertedString = "";
    s = s.toLowerCase();

    for(let i = 0; i < s.length; i++){
        if((s[i] >= 'a' && s[i] <= 'z') || (s[i] >= '0' && s[i] <= '9')){
            filtertedString += s[i];
        }
    }

    let left = 0;
    let right = filtertedString.length - 1;

    while(left < right){
        if(filtertedString[left] !== filtertedString[right]){
            return false;
        }
        left++;
        right--;
    }
    return true;
};

File: 125-valid-palindrome/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        // Convert to lowercase
        s = s.toLowerCase();

        // Build filtered string (only letters and digits)
        StringBuilder filtered = new StringBuilder();
        for (int i = 0; i < s.length(); i++) {
            char ch = s.charAt(i);
            if (Character.isLetterOrDigit(ch)) {
                filtered.append(ch);
            }
        }

        // Two-pointer check
        int left = 0;
        int right = filtered.length() - 1;
        while (left < right) {
            if (filtered.charAt(left) != filtered.charAt(right)) {
                return false;
            }
            left++;
            right--;
        }

        return true;
    }
}
